core/templates/pages/exploration-editor-page/services/exploration-rights-backend-api.service.ts

```diff
@@ -0,0 +1,193 @@
+      '/createhandler/rights/' + this.explorationDataService.explorationId);
+
+    return this.httpClient.put(requestUrl, {
+      version: this.explorationDataService.data.version,
+      viewable_if_private: viewableIfPrivate
+    }).toPromise().then((response: ExplorationRightsBackendResponse) => {
+      let data = response;
+      this.alertsService.clearWarnings();
+      that.init(
+        data.rights.owner_names, data.rights.editor_names,
+        data.rights.voice_artist_names, data.rights.viewer_names,
+        data.rights.status, data.rights.cloned_from,
+        data.rights.community_owned, data.rights.viewable_if_private);
+    });
+  }
+  saveRoleChanges(
+      newMemberUsername: string, newMemberRole: string): Promise<void> {
+    let that = this;
+    let requestUrl = (
+      '/createhandler/rights/' + this.explorationDataService.explorationId);
+
+    return this.httpClient.put(requestUrl, {
+      version: this.explorationDataService.data.version,
+      new_member_role: newMemberRole,
+      new_member_username: newMemberUsername
+    }).toPromise().then((response: ExplorationRightsBackendResponse) => {
+      let data = response;
+      that.alertsService.clearWarnings();
+      that.init(
+        data.rights.owner_names, data.rights.editor_names,
+        data.rights.voice_artist_names, data.rights.viewer_names,
+        data.rights.status, data.rights.cloned_from,
+        data.rights.community_owned, data.rights.viewable_if_private);
+    });
+  }
+  publish(): Promise<void> {
+    let that = this;
+    let requestUrl = (
+      '/createhandler/status/' + this.explorationDataService.explorationId);
+
+    return this.httpClient.put(requestUrl, {
+      make_public: true
+    }).toPromise().then((response: ExplorationRightsBackendResponse) => {
+      let data = response;
+      that.alertsService.clearWarnings();
+      that.init(
+        data.rights.owner_names, data.rights.editor_names,
+        data.rights.voice_artist_names, data.rights.viewer_names,
+        data.rights.status, data.rights.cloned_from,
+        data.rights.community_owned, data.rights.viewable_if_private);
+    });
+  }
+  saveModeratorChangeToBackend(emailBody: string): void {
+    let that = this;
+    let explorationModeratorRightsUrl = (
+      '/createhandler/moderatorrights/' +
+      this.explorationDataService.explorationId);
+
+    this.httpClient.put(explorationModeratorRightsUrl, {
+      email_body: emailBody,
+      version: this.explorationDataService.data.version
+    }).toPromise().then((response: ExplorationRightsBackendResponse) => {
+      let data = response;
+      that.alertsService.clearWarnings();
+      that.init(
+        data.rights.owner_names, data.rights.editor_names,
+        data.rights.voice_artist_names, data.rights.viewer_names,
+        data.rights.status, data.rights.cloned_from,
+        data.rights.community_owned, data.rights.viewable_if_private);
+    })['catch'](() => {
+      that.init(
```

Comment:
hm why not just do `that.init()`?

Reply:
It is because the first parameter of catch should be a function, but init does not return anything.